### PR TITLE
feat: builds binaries instead of storing them in the repo

### PR DESCRIPTION
**Pull request by kobigurk**
This PR introduces building the binaries directly instead of storing them in the repo. This helps with both version tracking and reproducibility of the build.

### CODE REVIEW

**Comment by kobigurk** [comment]
@silasbw I added a submodule here and also the necessary checkout config, but still doesn't seem to check out. Can you see why?

**Review by silasbw — COMMENTED** [review]
Seems OK to me.

Did you consider downloading pre-built / published binaries instead?

For example, setup a build + publish pipeline in https://github.com/celo-org/snark-setup that builds the binaries and publishes them to an http accessible location (e.g., azure blob) and then configure this repo/Dockerfile/code to download the prebuilt binaries (e.g., `wget https://azure/powersoftau-0.0.1.linux`).  That approach keeps the rust and typescript projects separate, avoids submodules, and might avoid some overhead in the CI snark-setup-coordinator.

Thoughts on that approach?

**Comment by kobigurk** [comment]
I've considered a similar approach and I think it's also fine.

The reason I opted for this one is clarity - it's still pretty
self-contained and allows a participant building everything from source
without going through hoops.

I'm happy to think more if you feel strongly about it.

File: coordinator-client/build_binaries.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+docker build -t snark-setup-binaries .
```
Comment (silasbw) [inline coordinator-client/build_binaries.sh:1]: Update the `coordinator-client/README.md` with instructions to run `sh build_binaries.sh`?

**Review by silasbw — APPROVED** [review]
I don't feel strongly, but I think having a greater degree of separation (something similar to what I proposed) offers some benefits.

File: coordinator-client/build_binaries.sh (hunk 1 above)
Comment (kobigurk) [reply coordinator-client/build_binaries.sh:1]: Will do!

**Pull request merged**